composer/benchmarker/benchmarker.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+from __future__ import annotations
+
+from typing import List, Optional, Sequence
+
+import torch
+import torch.distributed
+
+from composer.benchmarker.benchmarker_hparams import BenchmarkerHparams
+from composer.callbacks.benchmarker import Benchmarker as BenchmarkerCallback
+from composer.core.logging.base_backend import BaseLoggerBackend
+from composer.core.precision import Precision
+from composer.datasets.hparams import DataloaderSpec
+from composer.datasets.synthetic import SyntheticDataLabelType, SyntheticDataset
+from composer.models.base import BaseMosaicModel
+from composer.trainer.devices import DeviceCPU, DeviceGPU
+from composer.trainer.trainer import Trainer
+
+NUM_PROFILING_STEPS = 250  # needs to be high enough to be able to properly measure time
+
+
+class Benchmarker:
+    """Benchmarker for timing model training workloads with synthetic data.
+
+    Args:
+        model (BaseMosaicModel): The model to profile.
+        data_shape (List[int]): Shape of the tensor for input samples.
+        total_batch_size (int): The batch size to train with.
```

Comment:
See #103. `data_shape` may not work for NLP data.

Reply:
Yup - only meant to work for vision for now